core/include/log.h

```diff
@@ -49,12 +52,14 @@ extern "C" {
  * time, so a lower log level might result in smaller code size.
  */
 enum {
-    LOG_NONE,       /**< Lowest log level, will output nothing */
-    LOG_ERROR,      /**< Error log level, will print only critical,
-                         non-recoverable errors like hardware initialization
-                         failures */
+    LOG_NONE = -1,
```

Comment:
I think that changing this enum is a strong API change regarding the log level. In master, the debug log level corresponds to the value 4, with this change, it would become 7 if I'm not mistaken.

I can see that at least the `tests/log_color` test application won't work anymore as expected because of this:

https://github.com/RIOT-OS/RIOT/blob/2e2f950ca7ae0a455d48613c668dace842064a0c/tests/log_color/Makefile#L5-L6

There might be other applications affected as well.

Reply:
Would this really be an issue? I mean we can update the test apps (a proper way could be using `CFLAGS += -DLOG_LEVEL=LOG_DEBUG`). IMO any user code should use the enum and not the value directly...

Reply:
When directly used with CFLAGS, the enum can indeed be used directly and it's clearer.
There's also the corresponding Make variable `LOG_LEVEL` and its documentation that should be adapted:

https://github.com/RIOT-OS/RIOT/blob/bd5b195433d42ca5f29d348f7b967e609e33452a/makefiles/vars.inc.mk#L119

Reply:
Seems like the ESP32 SDK isn't too happy about this API change either :-/ https://ci.riot-os.org/RIOT-OS/RIOT/15739/c2c11e58d4921c710c5d776a686d3f239831829e/output/compile/tests/driver_apds99xx/esp32-ethernet-kit-v1_1:gnu.txt

Reply:
See my last commits for a workaround. 

If it's not ok with you, we can modify the ESP32 SDK, but I'm not sure what it exactly implies.